### PR TITLE
Load payment entity in IPN handler by transaction ID

### PR DESCRIPTION
**Pull request by krystalcode**
https://www.drupal.org/project/commerce_bluesnap/issues/3084996

### CODE REVIEW

**Comment by krystalcode** [comment]
@shabananavas Please test.

File: src/Ipn/Handler.php [hunk 1]
```diff
@@ -162,29 +162,30 @@ protected function getGroupType(array $ipn_data) {
...
       $this->logger->error($message);
       throw new BadRequestHttpException('Transaction ID required.');
     }
 
-    $payment = $this->entityTypeManager
-      ->getStorage('commerce_payment')
-      ->load($ipn_data['merchantTransactionId']);
-    if (!$payment) {
+    $payment_storage = $this->entityTypeManager->getStorage('commerce_payment');
+    $payment_id = $payment_storage->getQuery()
```
Comment (shabananavas) [inline src/Ipn/Handler.php]: @krystalcode Actually, `$payment_storage` has a function called loadByRemoteId(), so we can just do:

`$payment = $payment_storage->loadByRemoteId($ipn_data['referenceNumber']);`

**Comment by shabananavas** [comment]
@krystalcode I've posted my comments here: https://git.acromedia.com/accounts/bluesnap/issues/14

File: src/Ipn/Handler.php [hunk 2]
```diff
@@ -162,29 +162,30 @@ protected function getGroupType(array $ipn_data) {
...
+        'Payment with transaction ID "%s" not found for IPN of type "%s".',
+        $ipn_data['referenceNumber'],
         $this->getType($ipn_data)
       );
       $this->logger->error($message);
       throw new NotFoundHttpException('Transaction could not be found.');
     }
 
-    return $payment;
+    return $payment_storage->load($payment_id);
```
Comment (shabananavas) [inline src/Ipn/Handler.php]: $payment_id is an array anyway. So we're getting a Warning: array_flip(): Can only flip STRING and INTEGER values!

Comment (krystalcode) [reply src/Ipn/Handler.php]: Of course. Made the changes, should be good now - haven't tested myself though.

**Comment by shabananavas** [comment]
Looks good. These changes work as expected.

**Comment by krystalcode** [comment]
Merged on Drupal.org

**Pull request closed without merging**